### PR TITLE
libsail 0.9.0 (new formula)

### PR DESCRIPTION
**Pull request by HappySeaFox**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

SAIL is a C/C++ images decoding library similar to FreeImage, imlib2, or WIC. It's already available in Conan and vcpkg with thousands of installations. SAIL supports all modern image formats and provides simple API for end users.

### CODE REVIEW

File: Formula/lib/libsail.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          SAIL_TRY_OR_EXECUTE(sail_load_from_file(argv[1], &image),
+                                /* on error */ return 1);
+          sail_destroy_image(image);
+
+          return 0;
+      }
+    EOS
+
+    cflags = shell_output("#{Formula["pkg-config"].opt_bin}/pkg-config --cflags sail").strip.split
+    libs   = shell_output("#{Formula["pkg-config"].opt_bin}/pkg-config --libs sail").strip.split
```
[the remark this answers is not in the document]
Comment (HappySeaFox) [reply Formula/lib/libsail.rb:54]: The order of cflags and libs matters. On some platforms your proposal may work, on some may not. See https://stackoverflow.com/questions/15569850/order-of-parameters-in-g-clang-with-makefile

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/lib/libsail.rb:54]: I see, 👍

File: Formula/lib/libsail.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  depends_on "resvg"
+  depends_on "webp"
+
+  def install
+    args = %W[
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+      -DBUILD_TESTING=ON
+      -DSAIL_BUILD_EXAMPLES=OFF
+    ]
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, *args
```
Comment (chenrui333) [inline Formula/lib/libsail.rb:27]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
```

File: Formula/lib/libsail.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    system "ctest", "--test-dir", "build/tests", "--rerun-failed", "--output-on-failure"
```
Comment (chenrui333) [inline Formula/lib/libsail.rb:30]: no need to run tests per the release build, it should be part of the upstream pipeline already

```suggestion
```

File: Formula/lib/libsail.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    # To prevent conflicts with 'sail' formula
+    mv "#{bin}/sail", "#{bin}/sail-imaging"
```
Comment (chenrui333) [inline Formula/lib/libsail.rb:33]: 👍

**Review by chenrui333 — COMMENTED** [review]
Just some improvements, otherwise LGTM

**Comment by chenrui333** [comment]
@HappySeaFox can you squash the commits into one with commit msg `libsail 0.9.0 (new formula)` ? :)

**Review by chenrui333 — APPROVED** [review]
Thanks!

**Pull request merged**